examples/shader/blackhole.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build ignore
+// +build ignore
+
+package main
+
+var Time float
+var ScreenSize vec2
+var Cursor vec2
+
+func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
+	uv := position.xy / ScreenSize.xy
+	texC := uv
+	lpos := Cursor.xy / ScreenSize.x
+	texC2 := position.xy / ScreenSize.x
```

Comment:
Dividing the position (texel) by the screen size (pixel) doesn't make sense...

Perhaps you might want:

```
        pos := position.xy / imageDstTextureSize()
        origin, size := imageDstRegionOnTexture()
        pos -= origin
        pos /= size
```

See texel.go

Reply:
@hajimehoshi `position vec4` isn't pixels not pixels

But it makes sense using these functions, as for now it just works "by chance" if I'm correct about the potential offsets of `dst` on the offscreen ?

I'd go for
```go
origin, size := imageDstRegionOnTexture()
texC := (position.xy/imageDstTextureSize() - origin) / size
texC2 := (position.xy/imageDstTextureSize().x - origin) / size.x
lpos := Cursor.xy / ScreenSize.x
```

Reply:
> position vec4 isn't texels but pixels

You are right, sorry